### ISSUE TITLE
03. B Special header on demand

### ISSUE THREAD

**Issue by Jimmi08** [Jimmi08/e107-mdbootstrap-theme#4]
Combination of  Logo + Sitename
![image](https://user-images.githubusercontent.com/5429548/67636200-7916ce80-f8ce-11e9-83b6-cbe6d654fbfb.png)

and social buttons in header
![image](https://user-images.githubusercontent.com/5429548/67636206-9350ac80-f8ce-11e9-9214-ee7ab62ada8c.png)

**Comment by Jimmi08** [comment]
created file header_socials.html
if you want to use this one (without changing default), you need to rewrite shortcode
$header = varset( $this->customlayout['layout_header'] , "header_socials");	

With jmtheme plugin I can switch headers from admin area and I will not go back to hardcoded way :)

1. Added core {SITELOGO} shortcode - I dislike it because there is issue in Chrome console.

**Comment by Jimmi08** [comment]
Using sitelinks for social links: 
![image](https://user-images.githubusercontent.com/5429548/67637215-7325eb00-f8d8-11e9-9987-d8e9d0237116.png)

**Comment by Jimmi08** [comment]
https://github.com/Jimmi08/e107-mdbootstrap-theme/commit/c7b37b49637516a391654b8005813e28097ff09b

**Comment by Jimmi08** [comment]
If you want there icon with profile instead of avatar, then
replace this:
`{USER_AVATAR: w=35&h=35&crop=1} `  
with:
`<i class="fas fa-user"></i> Profile </a>`
in theme_shortcodes.php, userbox shortcode
![image](https://user-images.githubusercontent.com/5429548/67637267-fe9f7c00-f8d8-11e9-90cb-216fed27dd9d.png)

**Comment by DJJones66** [comment]
I do think when doing a MDBootstrap template you can use FontAwesome freely and everywhere since it is a given even in their own examples.  I think users almost expect to see it used

**Comment by DJJones66** [comment]
When using the sitelink, are you using fontawesome for your image or bootstrap's default image for like facebook?    Not that I have an issue, just curious how you are doing things

**Comment by Jimmi08** [comment]
Click on icon:
![image](https://user-images.githubusercontent.com/5429548/67639404-94de9c80-f8ef-11e9-99a0-ff88d19dec96.png)

Click on Tab Glyphs
look for correct glyphs
when you hover on icon, you will see name. You will need to be careful what icon you use -  if you select f.e. glyphicons, it will not be displayed.  


![image](https://user-images.githubusercontent.com/5429548/67639433-c6576800-f8ef-11e9-87d8-dafd989a5811.png)

You can too put code to name -  I would use this if the icon has a different name as it had in FA 4 
![image](https://user-images.githubusercontent.com/5429548/67639469-3cf46580-f8f0-11e9-9c05-129790570f2c.png)

There is a way how to adds icons to glyph manager (f.e. FA icons or other icon set), but I wasn't able to do it.

**Comment by DJJones66** [comment]
That was the same exact thing I did.. just wanted to make sure I was following the same :)

**Comment by Jimmi08** [comment]
if you notice latest changes to style.css, there was added trick for those icons - e107 uses fa (FA4) instead fab class (FA5)  
If any case when your icon is not displayed, you need to check the source code and then fix it.  Some fixes can be done just by javascript (see custom.js)  
Reason for this is that admin area uses bootstrap3 theme and it uses FA 4 icons, so Media Manager offers them and you could select wrong icon.

**Comment by DJJones66** [comment]
Yeah, I noticed the Admin was stuck on B3 and wondered if that could cause some issues down the road if not careful.

**Comment by Jimmi08** [comment]
True? If it was on me, I would stay with bootstrap3. Bootstrap4 is not the overkilling tool for CMS.   Except icons - no, I am not aware of any problems. other used bootstrap classes stayed the same.

**Comment by DJJones66** [comment]
From a coding perspective for me.. just being consistant, B4 is the norm now so keep updated or run the risk of larger problems down the road.  You see it happen with coders who do not feel updating something is important until it is to late.

**Comment by DJJones66** [comment]
On your wesbite.. do you have like a plugin package or something?

**Comment by Jimmi08** [comment]
What do you mean by like a plugin package?

B4 is just css framework, not the norm, you can use any others. It's about personal preferences.

**Comment by DJJones66** [comment]
Well, you have from what I can see just on the MB template you are doing 3 plugins... So was wondering if there was package for them or what?

**Comment by Jimmi08** [comment]
jmcore is free, you can download it from my repository
https://github.com/Jimmi08/jmcore-e107-plugin

JM Theme and JM Elements are paid.
https://www.e107sk.com/documentation/jm-elements-plugin/?cat.101
https://www.e107sk.com/documentation/jmtheme-plugin/?cat.98
You can see price at the store link.

Just delete them from theme.xml if they bother you. They just save time, you still can do everything manually with core stuff.

**Comment by DJJones66** [comment]
I found it.. price is in Euros correct?

**Comment by Jimmi08** [comment]
Yes, but I would wait to see them in action on this theme.  Maybe you don't need them.

**Comment by Jimmi08** [comment]
when we got to footer or other layouts (login, contact, signup) you will see power of JM theme.
when you will need content like features, timelines - any business stuff, not community -  you will see power of JM elements.

**Comment by DJJones66** [comment]
I am in the US.. Vermont

**Comment by Jimmi08** [comment]
I ask just because of that stupid VAT EU policy and paypal fees. Thanks

**Comment by DJJones66** [comment]
Yep.. life on the internet
haha

**Comment by Jimmi08** [comment]
I will be busy during week, but you could get some ideas about how your login screen should look:
I found this on MDB:
https://mdbootstrap.com/previews/templates/saas/html/pages/login.html
And footer
here is different that you have there now:
https://mdbootstrap.com/previews/templates/saas/html/home/home-onepage.html
this one is free:
https://mdbootstrap.com/freebies/jquery/portfolio/

It's not problem to change it later, just select one I could use for this version and you like it.

**Comment by DJJones66** [comment]
Really depends on how far you want to go.. but there are some things to be said for how clear their demo forms look

https://mdbootstrap.com/docs/jquery/forms/basic/

**Comment by DJJones66** [comment]
Me personally.. I would take that clean design.. and use tabs.. maybe I have a screenshot of it still
![GL_SignIn_01](https://user-images.githubusercontent.com/45010408/67641400-42de4c80-f8d8-11e9-8199-112c0677e756.jpg)

![GL_Register_01](https://user-images.githubusercontent.com/45010408/67641403-470a6a00-f8d8-11e9-931b-ae5232dc311b.jpg)

**Comment by DJJones66** [comment]
From one screen, cover it all...

**Comment by Jimmi08** [comment]
Try to find an example on MDB, it would be easier.  

I know this one:
https://agmstudio.io/themes/material-style/2.4.6/page-login_register.html

it's not easy but it can be done. I did it already long time ago for modal login on e107.org
But you have there terms and cookies...  

I will think about this.

**Comment by DJJones66** [comment]
When I did it.. I took the basic idea from here

https://mdbootstrap.com/docs/jquery/forms/basic/

Then expanded upon it

**Comment by Jimmi08** [comment]
Yes, but I would use source code, like they have this:
![image](https://user-images.githubusercontent.com/5429548/67641733-b39b5e00-f904-11e9-903f-0fc1fa5829c8.png)

Looking for nav tabs example...

**Comment by DJJones66** [comment]
This is the code I created for doing the testing of the idea...
<div class="container white z-depth-0" style="margin-top: 80px;max-width: 600px;">

    <div class="classic-tabs">

    <ul class="nav mdb-color lighten-1" id="myClassicTab" role="tablist">
        <li class="nav-item">
            <a class="nav-link  waves-light active show" id="profile-tab-classic" data-toggle="tab" href="#profile-classic"
               role="tab" aria-controls="profile-classic" aria-selected="true">Sign In</a>
        </li>
        <li class="nav-item">
            <a class="nav-link waves-light" id="follow-tab-classic" data-toggle="tab" href="#follow-classic" role="tab"
               aria-controls="follow-classic" aria-selected="false">Register</a>
        </li>
        <li class="nav-item">
            <a class="nav-link waves-light" id="contact-tab-classic" data-toggle="tab" href="#contact-classic" role="tab"
               aria-controls="contact-classic" aria-selected="false">Terms of Service</a>
        </li>
        <li class="nav-item">
            <a class="nav-link waves-light" id="awesome-tab-classic" data-toggle="tab" href="#awesome-classic" role="tab"
               aria-controls="awesome-classic" aria-selected="false">Cookies</a>
        </li>

    </ul>
    <div class="tab-content border-right border-bottom border-left rounded-bottom" id="myClassicTabContent">
        <div class="tab-pane fade active show" id="profile-classic" role="tabpanel" aria-labelledby="profile-tab-classic">
            
            <form name="SignInForm" id="SignInForm" class="text-center" style="color: #59698d ;margin-left: 10px;margin-right: 10px;"  action="<?php echo Settings::PATH['base'] ?>/gamers/signin" method="post" enctype="multipart/form-data">

                
                <div class="md-form">
                    <input type="text" id="materialLoginFormEmail" name="email"class="form-control" placeholder="Email" value="">
                    <label for="materialLoginFormEmail">E-mail</label>
                    <div class="error" id="Error_materialLoginFormEmail"></div>
                </div>

                
                <div class="md-form">
                    <input type="password" id="materialLoginFormPassword" name="password" class="form-control"  placeholder="Password" value="">
                    <label for="materialLoginFormPassword">Password</label>
                    <div class="error" id="Error_materialLoginFormPassword"></div>
                </div>

                <div class="d-flex justify-content-around">
                    <div>
                        
                        <div class="form-check">
                            <input type="checkbox" class="form-check-input" id="materialLoginFormRemember">
                            <label class="form-check-label" for="materialLoginFormRemember">Remember me</label>
                        </div>
                    </div>
                    <div>
                        
                        <a href="">Forgot password?</a>
                    </div>
                </div>

                
                <button class="btn btn-outline-info btn-rounded btn-block my-4 waves-effect z-depth-0 " style="color: #59698d !important;border: 2px solid #59698d !important;" type="submit">Sign in</button>

                
                <p>Not a member?
                    <a href="">Register</a>
                </p>



            </form>
            

        </div>
        <div class="tab-pane fade" id="follow-classic" role="tabpanel" aria-labelledby="follow-tab-classic">
            
            <form name="RegisterForm" id="RegisterForm" class="text-center" style="color: #59698d ;margin-left: 10px;margin-right: 10px;" action="gamers/register" method="post" enctype="multipart/form-data">
                <input id="id" name="id" type="hidden" class="form-control" value="0">
                <div class="row"  style="padding-left: 15px;">
                    <div class="md-form col s6"  style="padding-left: 0px;">
                        <input id="first_name" name="first_name" type="text" class="form-control">
                        <label for="first_name">First Name</label>
                        <div class="error" id="Error_first"></div>
                    </div>
                    <div class="md-form col s6"  style="padding-left: 0px;">
                        <input id="last_name" name="last_name" type="text" class="form-control">
                        <label for="last_name">Last Name</label>
                        <div class="error" id="Error_last"></div>
                    </div>
                </div>
                <div class="row"  style="padding-left: 15px;">
                    <div class="md-form col s12"   style="padding-left: 0px;">
                        <input id="email" name="email" type="email" class="form-control">
                        <label for="email">Email</label>
                        <div class="error" id="Error_email"></div>
                    </div>
                </div>
                <div class="row"  style="padding-left: 15px;">
                    <div class="md-form col s12"   style="padding-left: 0px;">
                        <input id="emailconfirm"  name="emailconfirm" type="email" class="form-control">
                        <label for="emailconfirm">Email Confirmation</label>
                        <div class="error" id="Error_emailConfirm"></div>
                    </div>
                </div>

                <div class="row"  style="padding-left: 15px;">
                    <div class="md-form col s12"   style="padding-left: 0px;">
                        <input id="password"  name="password" type="password" class="form-control">
                        <label for="password">Password</label>
                        <div class="error" id="Error_password"></div>
                    </div>
                </div>
                <div class="row"  style="padding-left: 15px;">
                    <div class="md-form col s12"   style="padding-left: 0px;">
                        <input id="passwordconfirm"  name="passwordconfirm" type="password" class="form-control">
                        <label for="passwordconfirm">Password Confirmation</label>
                        <div class="error" id="Error_passwordConfirm"></div>
                    </div>
                </div>

                <div class="d-flex justify-content-around">
                    <div>
                        
                        <div class="form-check">
                            <input type="checkbox" class="form-check-input" id="materialLoginFormTOS">
                            <label class="form-check-label" for="materialLoginFormTOS">Accept Terms of Service</label>
                            <div class="error" id="Error_"></div>
                        </div>
                    </div>
                    <div>
                        
                        <div class="form-check">
                            <input type="checkbox" class="form-check-input" id="materialLoginFormCookie">
                            <label class="form-check-label" for="materialLoginFormCookie">Accept Cookies</label>
                            <div class="error" id="Error_"></div>
                        </div>
                    </div>
                </div>




                
                <button id="RegisterFormButton" class="btn btn-outline-info btn-rounded btn-block my-4 waves-effect z-depth-0" style="color: #59698d !important;border: 2px solid #59698d !important;" type="submit">Register</button>





            </form>
            
        </div>
        <div class="tab-pane fade" id="contact-classic" role="tabpanel" aria-labelledby="contact-tab-classic">
            <p>Terms of service stuff </p>
        </div>
        <div class="tab-pane fade" id="awesome-classic" role="tabpanel" aria-labelledby="awesome-tab-classic">
            <p>This site uses cookes</p>
        </div>
    </div>

</div>

**Comment by DJJones66** [comment]
Seems not able to take the hole thing.. let me break it into 2 pieces
<div class="container white z-depth-0" style="margin-top: 80px;max-width: 600px;">

    <div class="classic-tabs">

    <ul class="nav mdb-color lighten-1" id="myClassicTab" role="tablist">
        <li class="nav-item">
            <a class="nav-link  waves-light active show" id="profile-tab-classic" data-toggle="tab" href="#profile-classic"
               role="tab" aria-controls="profile-classic" aria-selected="true">Sign In</a>
        </li>
        <li class="nav-item">
            <a class="nav-link waves-light" id="follow-tab-classic" data-toggle="tab" href="#follow-classic" role="tab"
               aria-controls="follow-classic" aria-selected="false">Register</a>
        </li>
        <li class="nav-item">
            <a class="nav-link waves-light" id="contact-tab-classic" data-toggle="tab" href="#contact-classic" role="tab"
               aria-controls="contact-classic" aria-selected="false">Terms of Service</a>
        </li>
        <li class="nav-item">
            <a class="nav-link waves-light" id="awesome-tab-classic" data-toggle="tab" href="#awesome-classic" role="tab"
               aria-controls="awesome-classic" aria-selected="false">Cookies</a>
        </li>

    </ul>
    <div class="tab-content border-right border-bottom border-left rounded-bottom" id="myClassicTabContent">
        <div class="tab-pane fade active show" id="profile-classic" role="tabpanel" aria-labelledby="profile-tab-classic">
            
            <form name="SignInForm" id="SignInForm" class="text-center" style="color: #59698d ;margin-left: 10px;margin-right: 10px;"  action="<?php echo Settings::PATH['base'] ?>/gamers/signin" method="post" enctype="multipart/form-data">

                
                <div class="md-form">
                    <input type="text" id="materialLoginFormEmail" name="email"class="form-control" placeholder="Email" value="">
                    <label for="materialLoginFormEmail">E-mail</label>
                    <div class="error" id="Error_materialLoginFormEmail"></div>
                </div>

                
                <div class="md-form">
                    <input type="password" id="materialLoginFormPassword" name="password" class="form-control"  placeholder="Password" value="">
                    <label for="materialLoginFormPassword">Password</label>
                    <div class="error" id="Error_materialLoginFormPassword"></div>
                </div>

                <div class="d-flex justify-content-around">
                    <div>
                        
                        <div class="form-check">
                            <input type="checkbox" class="form-check-input" id="materialLoginFormRemember">
                            <label class="form-check-label" for="materialLoginFormRemember">Remember me</label>
                        </div>
                    </div>
                    <div>
                        
                        <a href="">Forgot password?</a>
                    </div>
                </div>

                
                <button class="btn btn-outline-info btn-rounded btn-block my-4 waves-effect z-depth-0 " style="color: #59698d !important;border: 2px solid #59698d !important;" type="submit">Sign in</button>

                
                <p>Not a member?
                    <a href="">Register</a>
                </p>



            </form>
            

        </div>

**Comment by DJJones66** [comment]
<div class="tab-pane fade" id="follow-classic" role="tabpanel" aria-labelledby="follow-tab-classic">
            
            <form name="RegisterForm" id="RegisterForm" class="text-center" style="color: #59698d ;margin-left: 10px;margin-right: 10px;" action="gamers/register" method="post" enctype="multipart/form-data">
                <input id="id" name="id" type="hidden" class="form-control" value="0">
                <div class="row"  style="padding-left: 15px;">
                    <div class="md-form col s6"  style="padding-left: 0px;">
                        <input id="first_name" name="first_name" type="text" class="form-control">
                        <label for="first_name">First Name</label>
                        <div class="error" id="Error_first"></div>
                    </div>
                    <div class="md-form col s6"  style="padding-left: 0px;">
                        <input id="last_name" name="last_name" type="text" class="form-control">
                        <label for="last_name">Last Name</label>
                        <div class="error" id="Error_last"></div>
                    </div>
                </div>
                <div class="row"  style="padding-left: 15px;">
                    <div class="md-form col s12"   style="padding-left: 0px;">
                        <input id="email" name="email" type="email" class="form-control">
                        <label for="email">Email</label>
                        <div class="error" id="Error_email"></div>
                    </div>
                </div>
                <div class="row"  style="padding-left: 15px;">
                    <div class="md-form col s12"   style="padding-left: 0px;">
                        <input id="emailconfirm"  name="emailconfirm" type="email" class="form-control">
                        <label for="emailconfirm">Email Confirmation</label>
                        <div class="error" id="Error_emailConfirm"></div>
                    </div>
                </div>

                <div class="row"  style="padding-left: 15px;">
                    <div class="md-form col s12"   style="padding-left: 0px;">
                        <input id="password"  name="password" type="password" class="form-control">
                        <label for="password">Password</label>
                        <div class="error" id="Error_password"></div>
                    </div>
                </div>
                <div class="row"  style="padding-left: 15px;">
                    <div class="md-form col s12"   style="padding-left: 0px;">
                        <input id="passwordconfirm"  name="passwordconfirm" type="password" class="form-control">
                        <label for="passwordconfirm">Password Confirmation</label>
                        <div class="error" id="Error_passwordConfirm"></div>
                    </div>
                </div>

                <div class="d-flex justify-content-around">
                    <div>
                        
                        <div class="form-check">
                            <input type="checkbox" class="form-check-input" id="materialLoginFormTOS">
                            <label class="form-check-label" for="materialLoginFormTOS">Accept Terms of Service</label>
                            <div class="error" id="Error_"></div>
                        </div>
                    </div>
                    <div>
                        
                        <div class="form-check">
                            <input type="checkbox" class="form-check-input" id="materialLoginFormCookie">
                            <label class="form-check-label" for="materialLoginFormCookie">Accept Cookies</label>
                            <div class="error" id="Error_"></div>
                        </div>
                    </div>
                </div>




                
                <button id="RegisterFormButton" class="btn btn-outline-info btn-rounded btn-block my-4 waves-effect z-depth-0" style="color: #59698d !important;border: 2px solid #59698d !important;" type="submit">Register</button>





            </form>
            
        </div>
        <div class="tab-pane fade" id="contact-classic" role="tabpanel" aria-labelledby="contact-tab-classic">
            <p>Terms of service stuff </p>
        </div>
        <div class="tab-pane fade" id="awesome-classic" role="tabpanel" aria-labelledby="awesome-tab-classic">
            <p>This site uses cookes</p>
        </div>
    </div>

</div>

**Comment by DJJones66** [comment]
Odd.. seems to still be missing the first section of the div
Just the container div though

**Comment by Jimmi08** [comment]
Don't worry. I could use standard bootstrap 4 nav tabs solution at the beginning. 
What worry me is content - how to do it to be able to reuse it if I need it. 

JM Elements has already shortcode for separate signup form,
{SIGNUPFORM:  template=default}  

But I am afraid you will need both those plugins.  I will see.  I need to think.

**Comment by DJJones66** [comment]
No worries..

Good night and thanks

**Comment by Jimmi08** [comment]
Again, I knew that there are some limitations

- what about Forgotten password page?
- what about Coppa or how it's called when somebody register. 

But I really put my comp off now :)

**Comment by DJJones66** [comment]
Yeah will need to add those.. I just had not got that far when I realized I needed to change the path I was on at the time

**Comment by DJJones66** [comment]
Have you ever messed with doing custom templates for pages? Seems there is a default , custom and profile.. I think to do what I want can use pages if I can create my own templates.  For starters how to access the custom data you insert for pages?  Thank you

**Comment by DJJones66** [comment]
Found the page_template inside the theme :)  Now to figure out all the shortcodes, how to access data and such
